### PR TITLE
chore: Reserve capacity for vectors

### PR DESCRIPTION
**Pull request by relrelb**
This avoids unnecessary re-allocations and thus should improve performance in both memory and time terms.

### CODE REVIEW

File: core/src/display_object/edit_text.rs [hunk 1]
```diff
@@ -1583,11 +1583,6 @@ impl<'gc> TDisplayObject<'gc> for EditText<'gc> {
             .as_node()
             .duplicate(context.gc_context, true)
             .document();
-
-        let mut new_layout = Vec::new();
```
Comment (adrian17) [inline core/src/display_object/edit_text.rs:1587]: btw did you find out anything about this? Maybe there should be a note left about this somewhere...

**Review by adrian17 — APPROVED** [review]

File: core/src/display_object/edit_text.rs (hunk 1 above)
Comment (relrelb) [reply core/src/display_object/edit_text.rs:1587]: This code was introduced in d7a257f93faf8a585d1d00af02e1682014d9705e, and used to be:
```rs
text.layout = text.layout.map(|l| l.read().duplicate(context.gc_context));
```
which seems equivalent except the assignment to `text.layout`, that makes it effective.
@kmeisthax Maybe do you have an insight about this code change. Correct me if I'm wrong, but it feels to me that the previous code is still correct and should be re-introduced.

**Review by Herschel — APPROVED** [review]

**Comment by Herschel** [comment]
Thank you!

**Comment by SN902** [comment]
commit affects on https://github.com/ruffle-rs/ruffle/issues/2013 this game now run slower on PC and horribly on android

**Pull request merged**